timeless/views.py

```diff
@@ -49,151 +66,144 @@ def delete(self):
-            return redirect(url_for(self.not_found_url_name))
-
-        return self.render_template({"instance": instance})
+        return self.context_object_list_name
 
+    def get_object_list(self):
+        """
+        Get the list of objects. If this method is not overwritten, then a
+        model variable must be declared, and it must have query.all().
+        """
+        if self.model is None:
+            raise NotImplementedError(f"{self.__class__.__name__} must define "
+                                      f"either 'model' or 'get_object_list()'")
+        return self.model.query.all()
 
-class CreateUpdateView(SingleObjectMixin, GenericView):
-    """Example
-        class SettingsCreateUpdateViewView(views.CreateUpdateView):
-            template_name = "restaurants/tables/create_edit.html"
-            success_url_name = "reservation_settings_list"
-            form = forms.SettingsForm
-    """
-    form = None
-    success_url_name = None
-    not_found_url_name = None
-
-    def get(self, id=None):
-        """Get method render create template """
-        assert self.form, "Form is required"
-        form = self.form()
-        if id:
-            instance = self.get_object(id)
-            if not instance:
-                return redirect(url_for(self.not_found_url_name))
-            form = self.form.TableForm(instance=instance)
-
-        return self.render_template({"form": form})
-
-    def post(self, id=None):
+    def get_default_context(self):
         """
-        Post method checks form validation,
-        save into db and redirect to success_url
+        Add the object list to the context.
         """
-        assert self.form, "Form is required"
-        assert self.success_url_name, "Success_url_name is required"
-        form = self.form(request.form)
-        if form.validate():
-            form.save()
-        return redirect(url_for(self.success_url_name))
+        context = super().get_default_context()
+        context[self.get_context_object_list_name()] = self.get_object_list()
+        return context
 
 
-class DeleteView(SingleObjectMixin, GenericView):
+class DetailView(GenericView):
     """
-    Example:
-    class SettingsDeleteView(views.DeleteView):
-        success_url_name = "reservation_settings_list"
+    A view that will display details in a template for a single object.
     """
-    success_url_name = None
+    context_object_name = "object"
 
-    def delete(self, id=None):
+    def get_context_object_name(self):
         """
-        Calls the delete() method on the fetched object and then
-        redirects to the success URL.
+        Get context_object_name.
         """
-        instance = self.get_object(id)
-        instance.delete()
-        return redirect(url_for(self.success_url_name))
+        return self.context_object_name
 
-    def post(self, id=None):
-        """ Post method for object delete"""
-        return self.delete(id)
+    def get_object(self):
```

Comment:
@bkuzmic we know model and we know object id from url, so we can return object here. wdyt?

Reply:
@vryazanov My thought was that this is too much detail - not generic enough. What is someone wants to use the `username` instead of `id ` or some other unique field? I would like to leave this to the higher level of implementation.